tinygrad/nn/optim.py

```diff
@@ -8,8 +8,9 @@ def __init__(self, params: List[Tensor]):
     for x in params:
       if x.requires_grad is None: x.requires_grad = True
 
-    self.params: List[Tensor] = [x for x in params if x.requires_grad]
-    self.buffers: List[Tensor] = [x for x in params if not x.requires_grad]   # buffers are still realized
+    deduped_params: Set[Tensor] = set(params)
```

Comment:
Use `from tinygrad.helpers import dedup` instead. It preserves order and is `O(n)`